### PR TITLE
bignum support for hex and binary formats

### PR DESCRIPTION
**Pull request by kroggen**
This PR is for hardfork version 4

### CODE REVIEW

**Comment by kroggen** [comment]
TODO:
- [ ] add tests

File: contract/bignum_module.c [hunk 1]
```diff
@@ -67,7 +67,15 @@ const char *lua_set_bignum(lua_State *L, char *s) {
 	if (x == NULL) {
 		return mp_num_memory_error;
 	}
-	if (vm_is_hardfork(L, 3)) {
+	if (vm_is_hardfork(L, 4)) {
+		// remove support for octal format and
+		// keep support for hex (0x) and binary (0b) formats
+		if (s && s[0]=='0' && s[1]!=0 && s[1]!='x' && s[1]!='b') {
+			// convert "0123" -> "123"
+			while (s && s[0]=='0' && s[1]!=0) s++;
```
[the remark this answers is not in the document]
Comment (kroggen) [reply contract/bignum_module.c:75]: I will add these cases to the tests

[the remark this answers is not in the document]
Comment (kroggen) [reply contract/bignum_module.c:75]: @hayarobi the new test cases were added and are passing. single digits are being parsed correctly

**Review by hayarobi — APPROVED** [review]
lgtm

**Pull request merged**